forc-plugins/forc-crypto/src/content.rs

```diff
@@ -0,0 +1,63 @@
+use std::str::FromStr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::{
+    fs::read,
+    io::{self, Read},
+};
+
+static STDIN_HAS_BEEN_USED: AtomicBool = AtomicBool::new(false);
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Content {
+    Path(std::path::PathBuf, Vec<u8>),
+    Raw(Vec<u8>),
+}
+
+impl Content {
+    pub fn from_hex_or_raw(input: Vec<u8>) -> Vec<u8> {
+        if let Ok(text) = String::from_utf8(input.clone()) {
+            let text = text.trim();
+            if let Some(text) = text.strip_prefix("0x") {
+                if let Ok(bin) = hex::decode(text) {
+                    return bin;
+                }
+            }
+            text.as_bytes().to_vec()
+        } else {
+            input
+        }
+    }
+}
+
+impl FromStr for Content {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        if s == "-" {
+            if STDIN_HAS_BEEN_USED.load(Ordering::Acquire) {
+                let mut buffer = vec![];
+                let _ = io::stdin().lock().read_to_end(&mut buffer)?;
+                return Ok(Content::Raw(Self::from_hex_or_raw(buffer)));
+            }
+            STDIN_HAS_BEEN_USED.store(true, Ordering::SeqCst);
+            Ok(Content::Raw(vec![]))
+        } else {
+            let path = std::path::PathBuf::from(s);
```

Comment:
match statement might be a little cleaner here. 
```suggestion
            let path = PathBuf::from(s);
            match read(&path) {
                Ok(content) => Ok(Content::Path(path, Self::from_hex_or_raw(content))),
                Err(_) => Ok(Content::Raw(Self::from_hex_or_raw(s.as_bytes().to_vec()))),
            }
```